### ISSUE TITLE
MacOS tests fail to install Python 3.7 (which is EOL anyway)

### ISSUE THREAD

**Issue by mslw** [datalad/datalad#7584]
Our GitHub actions for MacOS testing use `macos-latest`, and it looks like they just started failing to set up Python 3.7: see https://github.com/datalad/datalad/actions/runs/8817812378/job/24205131401#step:4:11

```
Error: The version '3.7' with architecture 'arm64' was not found for macOS 14.4.1.
```

This was not the case 2 days ago, when the test ran on macOS 12.

Given that [3.7 reached EOL 10 months ago](https://endoflife.date/python), I think we need to bump the Python version rather than change the macOS version. For macOS tests for sure, maybe also for others if still used.

Related: #7400

**Comment by yarikoptic** [comment]
we ran into similar problem yesterday in dandi:
- https://github.com/dandi/dandi-schema/pull/236#issuecomment-2073840282
- https://github.com/dandi/dandi-schema/pull/238/files

we need to boost to at least 3.10 at least or stick to `macos-12`... I will propose a PR

**Comment by mslw** [comment]
Ha, I just noticed that `macos-latest` only provides 3.11 and 3.12 https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md#python 

Note that I already made a PR bumping all tests to 3.8, and I'm happy to tweak it further. Preferences whether to bump Python or lower macOS? I'd go with 3.11.

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`1.0.3`](https://github.com/datalad/datalad/releases/tag/1.0.3)
